### PR TITLE
bpo-46429: tweak deepfreeze output

### PR DESCRIPTION
**Pull request by kumaraditya303**
https://bugs.python.org/issue46429

- Generates better names for locals 
- Generates better names for ints

### CODE REVIEW

**Review by gvanrossum — COMMENTED** [review]
I don’t like the new macro. It hides things I’d like to see clearly. The const_str thing is nice though.

File: Tools/scripts/deepfreeze.py [hunk 1]
```diff
@@ -326,6 +329,10 @@ def _generate_int_for_bits(self, name: str, i: int, digit: int) -> None:
     def generate_int(self, name: str, i: int) -> str:
         if -5 <= i <= 256:
             return f"(PyObject *)&_PyLong_SMALL_INTS[_PY_NSMALLNEGINTS + {i}]"
+        if i > 0:
+            name = f"const_int_{i}"
+        else:
+            name = f"const_int_negative_{abs(i)}"
```
Comment (arhadthedev) [inline Tools/scripts/deepfreeze.py]: Will not it generate `const_int_negative_0`?

Comment (kumaraditya303) [reply Tools/scripts/deepfreeze.py]: ``-5 <= i <= 256`` uses cached integers so it is unreachable

Comment (gvanrossum) [reply Tools/scripts/deepfreeze.py]: Nevertheless `>= 0` reads better and avoids the question in the first place.
```suggestion
        if i >= 0:
            name = f"const_int_{i}"
        else:
            name = f"const_int_negative_{abs(i)}"
```

File: Tools/scripts/deepfreeze.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import re
 import time
 import types
+from string import ascii_letters
```
Comment (gvanrossum) [inline Tools/scripts/deepfreeze.py]: Won't need this if you use a regex.

File: Tools/scripts/deepfreeze.py [hunk 3]
```diff
@@ -170,6 +171,8 @@ def generate_bytes(self, name: str, b: bytes) -> str:
     def generate_unicode(self, name: str, s: str) -> str:
         if s in identifiers:
             return f"&_Py_ID({s})"
+        if all(i in ascii_letters or i == "_" or i.isdigit() for i in s):
```
Comment (gvanrossum) [inline Tools/scripts/deepfreeze.py]: A regex would look better here.
```suggestion
        if re.match(r"\A\w+\Z", s):
```

Comment (kumaraditya303) [reply Tools/scripts/deepfreeze.py]: Done

File: Tools/scripts/deepfreeze.py [hunk 4]
```diff
@@ -170,6 +170,8 @@ def generate_bytes(self, name: str, b: bytes) -> str:
     def generate_unicode(self, name: str, s: str) -> str:
         if s in identifiers:
             return f"&_Py_ID({s})"
+        if re.match(r'^[A-Za-z0-9_]+$', s):
```
Comment (gvanrossum) [inline Tools/scripts/deepfreeze.py]: Unfortunately this also matches an identifier followed by a `\n` (newline).

Comment (kumaraditya303) [inline Tools/scripts/deepfreeze.py]:
```suggestion
        if re.match(r'\A[A-Za-z0-9_]+\Z', s):
```

Comment (kumaraditya303) [reply Tools/scripts/deepfreeze.py]: Fixed now, I need to learn regex better :)

**Review by gvanrossum — APPROVED** [review]
Good!

**Pull request merged**